scripts/o3de/o3de/export_project.py

```diff
@@ -7,28 +7,69 @@
 #
 
 import argparse
+import fnmatch
+import glob
 import logging
 import os
 import pathlib
 import platform
-import sys
+import psutil
+import shutil
 
 from o3de import manifest, utils
+from typing import List
+
+# Account for some windows-specific attributes
+if platform.system().lower() == 'windows':
+    EXECUTABLE_EXTENSION = '.exe'
+    O3DE_SCRIPT_NAME = 'o3de.bat'
+    GENERATOR = None
+    CMAKE_GENERATOR_OPTIONS = ['-DLY_DISABLE_TEST_MODULES=ON']
+else:
+    EXECUTABLE_EXTENSION = ""
+    O3DE_SCRIPT_NAME = 'o3de.sh'
+    GENERATOR = "Ninja Multi-Config"
+    CMAKE_GENERATOR_OPTIONS = ['-DLY_DISABLE_TEST_MODULES=ON', '-DLY_STRIP_DEBUG_SYMBOLS=ON']
+
+# Regardless of the output package configuration, the tools used for the export process must be built with the profile configuration
+PREREQUISITE_TOOL_BUILD_CONFIG = "profile"
+
+
+class ExportProjectError(RuntimeError):
+    """
+    Define an error related to the runtime execution of export project so to handle messaging properly
+    """
+    pass
+
 
 class O3DEScriptExportContext(object):
     """
     The context object is used to store parameter values and variables throughout the lifetime of an export script's execution.
     It can also be passed onto nested scripts the export script may execute, which can in turn update the context as necessary.
     """
     
-    def __init__(self, export_script_path: pathlib.Path,
-                       project_path: pathlib.Path,
-                       engine_path: pathlib.Path,
-                       args: list = []) -> None:
+    def __init__(self,
+                 export_script_path: pathlib.Path,
+                 project_path: pathlib.Path,
+                 engine_path: pathlib.Path,
+                 args: list = [],
+                 cmake_additional_build_args: list = []) -> None:
+
         self._export_script_path = export_script_path
         self._project_path = project_path
         self._engine_path = engine_path
         self._args = args
+        self._cmake_additional_build_args = cmake_additional_build_args
+
+        project_json_data = manifest.get_project_json_data(project_path=project_path)
+        assert project_json_data, f"Invalid project configuration file '{project_path}/project.json'. Invalid settings."
+
+        project_name = project_json_data.get('project_name')
+        assert project_name, f"Invalid project configuration file '{project_path}/project.json'. 'project_name' not found in the settings"
+
+        self._project_name = project_name
```

Comment:
would it make sense to just include the project json data overall?

Reply:
Not really, unless there is a need. Its better to keep the pattern to the minimal information that is needed